File: ios/DetonatorGallery/Sources/DetonatorGallery/GalleryMediaReaderCloseRequest.swift
import Photos

import Detonator

public class GalleryMediaReaderCloseRequest: Request {
    public override func run() -> Void {
        let data: CloseData = decodeData()!
                
        GalleryMediaReaderReadRequest.fetchResults[data.id] = nil
                
        end()
    }
    
    public struct CloseData: Decodable {
        let id: Int
    }
}

File: android/DetonatorGallery/src/main/java/com/iconshot/detonator/gallery/GalleryMediaReaderCloseRequest.java
package com.iconshot.detonator.gallery;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.request.Request;

public class GalleryMediaReaderCloseRequest extends Request {
    public GalleryMediaReaderCloseRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        end();
    }
}
